### ISSUE TITLE
Wiki Link Create File quick fix creates a file without the .md extension

### ISSUE THREAD

**Issue by vsch** [vsch/idea-multimarkdown#133]
Introduced in the clean up merging of link code. As a workaround rename the file to have the .md extension, until the next release.

**Comment by breandan** [comment]
Looking forward to this feature!

**Comment by vsch** [comment]
@breandan, you are a man of modest desires. ;)

Just fixing another bug and will be releasing a version with a whole slew of fixes, today.

**Comment by vsch** [comment]
@breandan, it's out. With a few link refactoring fixes and bug list clean up. Wanted to get that out before I embark on adding full auto formatting of documents.

**Comment by breandan** [comment]
Is this available in 1.4.1? When I press ⌥⏎ to activate the [quick fix menu](http://www.jetbrains.org/intellij/sdk/docs/tutorials/custom_language_support/quick_fix.html) on an Unresolved Link Reference, nothing happens. Running on IU-143.1184.

**Comment by vsch** [comment]
I am running 144.2608 and tried it on 143.1559 and on windows. You may need to invalidate caches and wait till they rebuild.

<img width="1078" alt="screen shot 2016-01-07 at 8 34 24 pm" src="https://cloud.githubusercontent.com/assets/10299086/12187906/13672e50-b57e-11e5-938a-b4182ea1bb62.png">

**Comment by vsch** [comment]
Just realized I fixed a bug today that under some conditions links would not highlight them unresolved, but if it highlights unresolved then it should offer create file quick fix.

**Comment by vsch** [comment]
Installed and tried 1.4.1 and it works. Do you get any exceptions?

<img width="1076" alt="screen shot 2016-01-07 at 8 40 49 pm" src="https://cloud.githubusercontent.com/assets/10299086/12187981/ff837762-b57e-11e5-8c11-4b16da2dab52.png">

**Comment by breandan** [comment]
Tried invalidating the cache and restarting, no such luck. Nothing happens when I invoke the intentions menu:

![image](https://cloud.githubusercontent.com/assets/175716/12188541/17a0492e-b56b-11e5-9876-ed32ca660cf9.png)

**Comment by vsch** [comment]
@breandan, interesting. 

1. According to the last exception in the log, dated: 2016-01-07 00:37:40 You are still running 1.4.0.2 not 1.4.1, since the lines in the trace correspond to 1.4.0 branch of code. The changes made in 1.4.1 affect the line numbers involved in the stack trace.

    Can you verify please verify that it is indeed 1.4.1 you are running. Don't know if the last exception happened before the upgrade or after.

2. I also discovered where I forgot to get rid of native thread creation: in the HTML Preview tabs. It is causing exceptions on your system but I don't think it is related to the lack of create file fix but it could. Causes your system to run out of native threads. I'll address this in the next release.

    Please go into preferences and uncheck Editor Tabs: Preview option. You don't need it since you are running split screen. This will eliminate unnecessary resource usage until it is fixed. I will also add a change to turn off preview tab for licensed versions. It does that in the new code when the license is activated but anyone who ran previous versions already has the settings saved and the license activated, leaving it there by default.

    <img width="971" alt="screen shot 2016-01-08 at 1 07 04 am" src="https://cloud.githubusercontent.com/assets/10299086/12192100/db5d92d4-b5a9-11e5-9d8d-55f2549b48ca.png">

3. You are also getting exceptions for taskitems.ttf font that is used the book icon, links on hover over headers, and open and closed task list items in JavaFX preview, see trace below:

    Is this an issue caused by corrupted font in the distribution or something on your system?

    The reason I ask is I made the mistake of trying git-lfs on my repo before reading the fine-print: it does not work with OS X apps without environment configuration. Only command line. So I was getting binary files truncated on checkout. Just wonder if there were other side-effects.

    ```text
2016-01-07 02:02:06,536 [5046524]   INFO -                     #GitStatus - git-lfs clean 'src/main/resources/com/vladsch/idea/multimarkdown/taskitems.ttf': git-lfs: command not found
error: cannot feed the input to external filter git-lfs clean %f
error: external filter git-lfs clean %f failed -1
error: external filter git-lfs clean %f failed
fatal: src/main/resources/com/vladsch/idea/multimarkdown/taskitems.ttf: clean filter 'lfs' failed
 during executing git -c core.quotepath=false status --porcelain -z --untracked-files=no -- src/main/java/com/vladsch/idea/multimarkdown/template/MultiMarkdownLiveTemplateProvider.java 
com.intellij.openapi.vcs.VcsException: git-lfs clean 
    ```

Amazing what real data from users will teach you about your own code.

**Comment by breandan** [comment]
1. That is strange. I did verify its running [1.4.1 on IntelliJ IDEA](https://cloud.githubusercontent.com/assets/175716/12193025/25be9f14-b599-11e5-8be7-5497d12fb78e.png), but just to be sure I reinstalled it from scratch on a fresh instance of RubyMine and WebStorm. Same issue. I've attached [the RubyMine log](https://github.com/vsch/idea-multimarkdown/files/82362/rubymine.txt), unfortunately it doesn't look very helpful. Let me know if you need anything else to diagnose.

2. Good to know, thanks. I *did* notice some increased sluggishness, but I wasn't sure if it was my old ThinkPad or some plugin.

3. Not sure what's going on there, the fonts look fine to me.

**Comment by vsch** [comment]
@breandan, my bad. It is a Windows thing. I am debugging it now.

**Comment by vsch** [comment]
@breandan, fixed. Path differences under windows, when I fixed the last bug, it propagated the problem down the road. Need to add windows based paths to my tests.

This build also has a fix for edit injected fragment opening an editor with only the top tab showing and not resizeable.

<https://github.com/vsch/idea-multimarkdown/raw/master/dist/idea-multimarkdown.1.4.1.1.zip>

**Comment by breandan** [comment]
It works! Awesome.

**Comment by vsch** [comment]
Next time complain sooner. :) I did not realize there was a problem on Windows. 

I now have my Windows test and PITA debug environment setup. Have to take the time to make it into a real dev environment, but the key mapping differences are killing me. I am slowly changing my Mac keymap on Windows so it is the same keys. I never use ALT-menu access on windows. I just wish there was a fast way to say META in the imported keymap means ALT after import. Tried editing the file manually but it did not work and I didn't have the time to figure it out.

**Comment by vsch** [comment]
@breandan, i made a build for fix 142, it also fixes for link refactoring. 

I also moved the first three annotation quick fixes to intentions: the header format refactoring ones, just to learn what's involved. Sweet. I will have to add some documentation to the SDK in the intention section while it is still fresh in my head. The section on intentions is miserly. Almost useless. I found what I needed in inteillij-community, but some docs to clarify what goes where and what it means would have been very helpful.

Build 1.4.1.3 <https://github.com/vsch/idea-multimarkdown/raw/master/dist/idea-multimarkdown.1.4.1.3.zip>

This build has **file move refactoring of links contained in the file**. Refactoring links targeting the file was done long ago, but I came across a mother load of good examples while looking for intention samples. Now link refactoring is pretty much done. I just love developing with and for the IntelliJ platform.

As an aside, I jotted down a page or two of ideas for a future project and used the plugin for the docs. Links, images, anchor refs, renaming header text so the anchors stay in sync. Used completions for links and link text. Loved it. This is the plugin I was looking for before I wound up developing it myself.

It may sound funny, but I get to spend a lot of time with it, mostly testing and debugging. I rarely get to just lay back enjoy using it.
